src/Slider.js

```diff
@@ -1,80 +1,30 @@
-import React, { Component } from 'react';
+import React, { useEffect, useRef } from 'react';
 import PropTypes from 'prop-types';
 import cx from 'classnames';
 
-class Slider extends Component {
-  constructor(props) {
-    super(props);
-    this.initSlider = this.initSlider.bind(this);
-    this.fullscreenReset = this.fullscreenReset.bind(this);
-    this.setActiveIndex = this.setActiveIndex.bind(this);
-  }
-
-  initSlider() {
-    if (typeof M !== 'undefined') {
-      this.instance = M.Slider.init(this._slider, this.props.options);
-    }
-  }
-
-  /**
-   * If the slider was not in fullscreen, the height is set as a style attribute
-   * on the Slider element. When `.destroy()` is called, this attribute is not
-   * removed, resulting in a fullscreen displayed incorrectly.
-   */
-  fullscreenReset(wasFullscreen) {
-    if (!wasFullscreen && this.props.fullscreen) {
-      this.instance.el.removeAttribute('style');
-      this.instance.el.childNodes[0].removeAttribute('style');
-    }
-  }
-
-  setActiveIndex(activeIndex) {
-    let { indicators } = this.props.options;
-    // In case this option is not defined, we assume true, as per default
-    let showIndicators = typeof indicators === 'undefined' || indicators;
-    if (showIndicators && activeIndex) {
-      this.instance['$indicators'][activeIndex].className =
-        'indicator-item active';
-    }
-  }
-
-  componentDidMount() {
-    this.initSlider();
-  }
-
-  componentDidUpdate(prevProps) {
-    if (!this.instance) return;
-    const activeIndex = this.instance.activeIndex;
-    this.instance.destroy();
-    this.fullscreenReset(prevProps.fullscreen);
-    this.initSlider();
-    // keep indicator at current index displayed
-    this.setActiveIndex(activeIndex);
-  }
+const Slider = ({ children, className, options, fullscreen, ...props }) => {
+  const _slider = useRef(null);
 
-  componentWillUnmount() {
-    if (this.instance) {
-      this.instance.destroy();
-    }
-  }
+  useEffect(() => {
+    const instance = M.Slider.init(_slider.current, options);
 
-  render() {
-    const { fullscreen, children, className } = this.props;
-    const classes = {
-      fullscreen,
-      slider: true
+    return () => {
+      instance.current && instance.current.destroy();
```

Comment:
this is just `instance && instance.destroy()`

Reply:
Whoops, my bad